fix: 'settings' is not defined. 문제 해결

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #80 

### 📌 설명

현재 `ChannelPlayer`의 `settings`에서 `'settings' is not defined.` 에러가 발생하고 있습니다. 또한, `Home` 페이지에서 라디오 채널 목록에 라디오 채널을 클릭했을 때, `'settings' is not defined.` 에러가 발생하여 서비스를 정상적으로 사용할 수 없는 문제를 해결한 Pull Request입니다.

### 📃 작업 사항

- [x] 'settings' is not defined. 문제 해결 

### 💡 참고 사항

`useChannelPlayback`에서 `useUserStore`의 `settings` 정보를 다루고 있으므로 반환값에 `settings`를 추가해주고 `ChannelPlayer` 컴포넌트에서 해당 반환값을 가져오는 방식으로 해결했습니다.

### 💭 리뷰 사항 반영

- [x] `ChannelPlayer`처럼 실제로 `settings`를 사용하는 곳에서 `useUserStore`를 직접 호출하는 방식으로 변경

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
fix: 'settings' is not defined. 문제 해결

useChannelPlayback에서 useUserStore의 settings 정보를 다루고 있으므로 반환값에 settings를 추가해주고
ChannelPlayer 컴포넌트에서 해당 반환값을 가져오는 방식으로 해결

faber-h commented on src/hooks/useChannelPlayback.jsx:
```diff
@@ -26,6 +26,7 @@ const useChannelPlayback = () => {
     videoId,
     selectedChannel,
     isPlaying,
+    settings,
```
miniPlayer에서는 settings 값을 사용하지 않기 때문에,
useChannelPlayback에서 함께 반환하게 되면 해당 컴포넌트에는 불필요한 데이터가 전달될 수 있을 것 같아요!

그래서 ChannelPlayer처럼 실제로 settings를 사용하는 곳에서
useUserStore를 직접 호출하는 방식이 관심사 분리 측면이나 재사용성 측면에서 더 깔끔할 것 같아요~

@Eun0713 @devjaeyoon 
두 분의 의견이 궁금합니다!

d0-neii replied:
희주님 말씀에 동의해요!
`settings`는 실제로 사용하는 컴포넌트에서 호출하는 게 관심사 분리 측면에서 더 깔끔한 구조인 것 같고 유지보수성 측면에서도 더 나은 방식인 것 같습니다🤗

devjaeyoon replied:
희주님의 의견에 동의합니다! `ChannelPlayer`는 결과적으로 `useUserStore`를 두 번 호출하게 되지만 `settings`를 사용하지 않는 `MiniPlayer` 보단 더 좋은 방식이라고 생각합니다! 해당 사항 반영하겠습니다!

devjaeyoon committed:
refactor: useChannelPlayback에서 settings 반환값 제거 및 ChannelPlayer에서 사용으로 변경

useChannelPlayback 훅에서는 settings를 사용하는 컴포넌트가 ChannelPlayer 하나뿐이고,
해당 값이 다른 곳에서는 불필요하게 반환되고 있었기 때문에 반환값에서 제거.
ChannelPlayer에서는 useUserStore를 통해 직접 settings를 가져오도록 변경하여
훅의 책임을 명확히 하고 불필요한 의존성을 줄임.

faber-h reviewed (approved):
LGTM 🔥

Eun0713 reviewed (approved):
LGTM🌈
